modified memcached operator according to run bundle integration (#86)
operator-framework/java-operator-plugins: docs/tutorial.md

If you modify the size field of the `memcached-sample.yaml` and re-apply it. The
operator will trigger a reconcile and adjust the sample pods to the size given.

### Deploy your Operator with OLM
First, install [OLM](https://sdk.operatorframework.io/docs/olm-integration/tutorial-bundle/#enabling-olm):

```
operator-sdk olm install
```

> Bundle your operator, then build and push the bundle image. The [bundle](https://github.com/operator-framework/operator-registry/blob/v1.16.1/docs/design/operator-bundle.md#operator-bundle) target generates a bundle in the bundle directory containing manifests and metadata defining your operator. bundle-build and bundle-push build and push a bundle image defined by bundle.Dockerfile.

Reviewer: ```suggestion
Bundle your operator, then build and push the bundle image. The [bundle](https://github.com/operator-framework/operator-registry/blob/v1.16.1/docs/design/operator-bundle.md#operator-bundle) target generates a bundle in the `bundle` directory containing manifests and metadata defining your operator. `bundle-build` and `bundle-push` build and push a bundle image defined by `bundle.Dockerfile`.
```

Author: Yes, Updated.

> Bundle your operator, then build and push the bundle image. The [bundle](https://github.com/operator-framework/operator-registry/blob/v1.16.1/docs/design/operator-bundle.md#operator-bundle) target generates a bundle in the bundle directory containing manifests and metadata defining your operator. bundle-build and bundle-push build and push a bundle image defined by bundle.Dockerfile.

Reviewer 2: Should we reference the latest release bundle documentation?

```suggestion
Bundle your operator, then build and push the bundle image. The [bundle](https://github.com/operator-framework/operator-registry/blob/v1.23.0/docs/design/operator-bundle.md#operator-bundle) target generates a bundle in the bundle directory containing manifests and metadata defining your operator. bundle-build and bundle-push build and push a bundle image defined by bundle.Dockerfile.
```

Author: Yes right. updated it.

```

> make bundle bundle-build bundle-push

Reviewer: Do we not have to specify the bundle image via `BUNDLE_IMG`? Or does will it magically create `memcached-operator-bundle`?

Author: Yes, either User needs to pass it or modify the makefile. The below link is showing how to pass it
https://sdk.operatorframework.io/docs/olm-integration/quickstart-bundle/#steps and I added that in the tutorial.

Author: Maybe we can ask User to export below variables

```
$ export USERNAME=<container-registry-username>
$ export VERSION=0.0.1
$ export IMG=docker.io/$USERNAME/memcached-operator:v$VERSION // location where your operator image is hosted
$ export BUNDLE_IMG=docker.io/$USERNAME/memcached-operator-bundle:v$VERSION // location where your bundle will be hosted
```

I will add it so that it will help User.

```

> Finally, run your bundle. If your bundle image is hosted in a registry that is private and/or has a custom CA, these [configuration steps](https://sdk.operatorframework.io/docs/olm-integration/cli-overview/#private-bundle-and-catalog-image-registries) must be complete.

Reviewer: ```suggestion
Finally, run your bundle. If your bundle image is hosted in a registry that is private and/or has a custom CA, these [configuration steps](https://sdk.operatorframework.io/docs/olm-integration/cli-overview/#private-bundle-and-catalog-image-registries) must be completed.
```

Author: Updated as per suggestion.

```
operator-sdk run bundle <some-registry>/memcached-operator-bundle:v0.0.1

> ```

Reviewer: I feel like we should show them some of the output for run bundle. So they know what to look for. But since what you have is consistent with the other tutorials. No need to change this.

Author: Yes right. As per me, we should show then the output lines of the run bundle. If any user who is not familiar with go lang and only looking for java then they can understand it.

I will add the below lines at the end

```
INFO[0009] Successfully created registry pod: docker-io-013859989-memcached-quarkus-operator-bundle-v0-1-1 
INFO[0009] Created CatalogSource: memcached-quarkus-operator-catalog 
INFO[0009] OperatorGroup "operator-sdk-og" created      
INFO[0009] Created Subscription: memcached-quarkus-operator-v0-1-1-sub 
INFO[0013] Approved InstallPlan install-6n8vm for the Subscription: memcached-quarkus-operator-v0-1-1-sub 
INFO[0013] Waiting for ClusterServiceVersion "default/memcached-quarkus-operator.v0.1.1" to reach 'Succeeded' phase 
INFO[0013]   Waiting for ClusterServiceVersion "default/memcached-quarkus-operator.v0.1.1" to appear 
INFO[0020]   Found ClusterServiceVersion "default/memcached-quarkus-operator.v0.1.1" phase: Pending 
INFO[0021]   Found ClusterServiceVersion "default/memcached-quarkus-operator.v0.1.1" phase: Installing 
INFO[0051]   Found ClusterServiceVersion "default/memcached-quarkus-operator.v0.1.1" phase: Succeeded 
INFO[0051] OLM has successfully installed "memcached-quarkus-operator.v0.1.1" 
```
